docs/features/software-catalog/descriptor-format.md

```diff
@@ -0,0 +1,295 @@
+# Descriptor Format of Catalog Entities
+
+This section describes the default data shape and semantics of catalog entities.
+
+This both applies to objects given to and returned from the software catalog API,
+as well as to the descriptor files that the software catalog can ingest natively. In
+the API request/response cycle, a JSON representation is used, while the descriptor
+files are on YAML format to be more easily maintainable by humans. However, the
+structure and semantics is the same in both cases.
+
+## Contents
+
+- [Overall Shape Of An Entity](#overall-shape-of-an-entity)
+- [Common to All Kinds: The Envelope](#common-to-all-kinds-the-envelope)
+- [Common to All Kinds: The Metadata](#common-to-all-kinds-the-metadata)
+- [Kind: Component](#kind-component)
+
+## Overall Shape Of An Entity
+
+The following is an example of the shape of an entity as returned from the software
+catalog API.
+
+```js
+{
+  "apiVersion": "backstage.io/v1beta1",
+  "kind": "Component",
+  "metadata": {
+    "annotations": {
+      "backstage.io/managed-by-location": "file:/tmp/component-info.yaml",
+      "example.com/service-discovery": "artistweb",
+      "circleci.com/project-slug": "gh/example-org/artist-website"
+    },
+    "description": "The place to be, for great artists",
+    "etag": "ZjU2MWRkZWUtMmMxZS00YTZiLWFmMWMtOTE1NGNiZDdlYzNk",
+    "generation": 1,
+    "labels": {
+      "system": "public-websites"
+    },
+    "name": "artist-web",
+    "uid": "2152f463-549d-4d8d-a94d-ce2b7676c6e2"
+  },
+  "spec": {
+    "lifecycle": "production",
+    "owner": "artist-relations@example.com",
+    "type": "website"
+  }
+}
+```
+
+The corresponding descriptor file that generated it may look as follows:
+
+```yaml
+apiVersion: backstage.io/v1beta1
+kind: Component
+metadata:
+  name: artist-web
+  description: The place to be, for great artists
+  labels:
+    system: public-websites
+  annotations:
+    example.com/service-discovery: artistweb
+    circleci.com/project-slug: gh/example-org/artist-website
+spec:
+  type: website
+  lifecycle: production
+  owner: artist-relations@example.com
+```
+
+The root fields `apiVersion`, `kind`, `metadata`, and `spec` are part of the
+_envelope_, defining the overall structure of all kinds of entity. Likewise,
+some metadata fields like `name`, `labels`, and `annotations` are of
+special significance and have reserved purposes and distinct shapes.
+
+See below for details about these fields.
+
+## Common to All Kinds: The Envelope
+
+The root envelope object has the following structure.
+
+### `apiVersion` and `kind` [required]
+
+The `kind` is the high level entity type being described. The
+[Backstage system model](https://github.com/spotify/backstage/issues/390) describes
```

Comment:
This issue will be closed at some point. I would prefer if you linked to a permanent location for this information, an ADR for example.

Also, should we not specify examples of `Kind`s that are supported now? Maybe link to the paragraph further down.

Reply:
@stefanalund OK, pointed to ADR005 instead and linked to the Component section.